llvm/lib/Transforms/Vectorize/VPlanVerifier.cpp

```diff
@@ -92,7 +92,47 @@ static bool verifyVPBasicBlock(const VPBasicBlock *VPBB,
   for (const VPRecipeBase &R : *VPBB)
     RecipeNumbering[&R] = Cnt++;
 
+  // Set of recipe types along with VPInstruction Opcodes of all EVL-related
+  // recipes that must appear at most once in Entry or Exiting blocks.
+  DenseSet<unsigned> EVLFound;
+  const VPlan *Plan = VPBB->getPlan();
+  bool IsHeader = Plan && Plan->getEntry()->getNumSuccessors() == 1 &&
```

Comment:
Plan must always be set here I think? Why do we need to check the number of successors here?

Reply:
I answered this question before, some of the unit tests crash without this check, they have recipes/vp-blocks without plan